necromorphs/code/modules/weapons/guns/pulse_rifle.dm

```diff
@@ -0,0 +1,192 @@
+/**
+Pulse Rifles
+*/
+
+/obj/item/gun/ballistic/automatic/pulse
+	name = "SWS Motorized Pulse Rifle"
+	desc = "The SWS Motorized Pulse Rifle is a military-grade, triple-barreled assault rifle, manufactured by Winchester Arms, is capable of a rapid rate of fire. \
+			The Pulse Rifle is the standard-issue service rifle of the Earth Defense Force and is also common among corporate security officers. "
+	icon = 'necromorphs/icons/obj/weapons/ds13guns48x32.dmi'
+	icon_state = "pulserifle"
+	lefthand_file = 'necromorphs/icons/mob/onmob/items/lefthand_guns.dmi'
+	righthand_file = 'necromorphs/icons/mob/onmob/items/righthand_guns.dmi'
+	inhand_icon_state = "pulserifle"
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = ITEM_SLOT_BELT|ITEM_SLOT_BACK|ITEM_SLOT_SUITSTORE
+	mag_display = FALSE
+	mag_type = /obj/item/ammo_box/magazine/pulse
+	weapon_weight = WEAPON_HEAVY
+	show_bolt_icon = FALSE
+	burst_size = 3//Old minimum burst was 2. Also, burst fire stacks with the automatic mode, meaning mag dump 3 times faster, in nearly 3 seconds
+	bolt_type = BOLT_TYPE_OPEN
+	can_suppress = FALSE
+	spread = 5
+	//firing_burst = TRUE  //Does this even work?
+	fire_sound = 'sound/ds13/pulse_shot.ogg'
+	load_sound = 'sound/ds13/pulse_magin.ogg'
+	eject_sound = 'sound/ds13/pulse_magout.ogg'
+
+/obj/item/gun/ballistic/automatic/pulse/no_mag
+	spawnwithmagazine = FALSE
+
+/obj/item/gun/ballistic/automatic/pulse/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/automatic_fire, 0.1 SECONDS)
+
+/obj/item/gun/ballistic/automatic/pulse/egov //Same situation as rending divet
+	name = "SWS Earthgov Motorized Pulse Rifle"
+	desc = "The SWS Motorized Pulse Rifle is a military-grade, triple-barreled assault rifle, manufactured by Winchester Arms, is capable of a rapid rate of fire. \
+This variant is the earthgov standard, featuring the highest grade parts. "
+	icon_state = "pulserifle_egov"
+	inhand_icon_state = "pulserifle_egov"
+	projectile_damage_multiplier = 1.10
+	//tier_1_bonus = 0
+
+/obj/item/gun/ballistic/automatic/pulse/egov/no_mag
+	spawnwithmagazine = FALSE
+
+/**
+Magazines
+*/
+
+/obj/item/ammo_box/magazine/pulse
+	name = "pulse magazine (standard)"
+	desc = "With a distinctive \"bell and stock\" design, pulse magazines can be inserted and removed from the Pulse Rifle with minimal effort and risk."
+	icon = 'necromorphs/icons/obj/ammo.dmi'
+	icon_state = "pulse_rounds"
+	caliber = CALIBER_PULSE
+	ammo_type = /obj/item/ammo_casing/caseless/pulse
+	max_ammo = 50 //To low? 65 normal
+
+// /obj/item/ammo_box/magazine/pulse/update_icon_state()
+// 	..()
+// 	if (length(stored_ammo))
+// 		icon_state = "pulse_rounds"
+// 	else
+// 		icon_state = "pulse_rounds_empty"
```

Comment:
Why did you leave these comments? If the code is not used and you aren't going to implement it in the future - remove it